### PR TITLE
Better encapsulate dispatcher.Message. Change gRPC payload from string to Message

### LINKED ISSUE #33: Change gRPC dispatcher to use byte[]+headers
See https://github.com/projectriff/function-proto/issues/5

### LINKED ISSUE #34: Better encapsulate dispatcher.Message
Also, change type of headers to `map[string][]string ie MultiMap<String, String>`

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #33
Fixes #34

### CODE REVIEW

File: pkg/dispatcher/grpc/fntypes/fntypes.proto [hunk 1]
```diff
@@ -14,21 +14,14 @@
...
-    oneof type {
-	    string body = 1;
-	    google.protobuf.Empty void = 2;
+message Message {
+	message HeaderValue {
+		repeated string values = 1;
 	}
+
+    bytes payload = 1;
+	map<string, HeaderValue> headers = 2;
```
Comment (markfisher) [inline pkg/dispatcher/grpc/fntypes/fntypes.proto]: fix indentation

Comment (dturanski) [inline pkg/dispatcher/grpc/fntypes/fntypes.proto]: Should we delete this from source code control since the build downloads it from function-proto?

Comment (markfisher) [reply pkg/dispatcher/grpc/fntypes/fntypes.proto]: @ericbottard ^^

Comment (ericbottard) [reply pkg/dispatcher/grpc/fntypes/fntypes.proto]: We could, although it's not exactly the approach we have we the whole `vendor/` directory for example. The makefile would need tweaking, as it currently doesn't trigger the download

Comment (markfisher) [reply pkg/dispatcher/grpc/fntypes/fntypes.proto]: fixed on merge

**Comment by markfisher** [comment]
rebased and merged as https://github.com/projectriff/function-sidecar/commit/bec62989766783fa3be6fcece8e8401fa1dabf62

**Pull request closed without merging**